domain/secret/rotatepolicy.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secret
+
+import coresecrets "github.com/juju/juju/core/secrets"
+
+// RotatePolicy represents the rotate policy of a secret.
+// as recorded in the secret_rotate_policy lookup table.
+type RotatePolicy int
+
+const (
+	RotateNever RotatePolicy = iota
+	RotateHourly
+	RotateDaily
+	RotateWeekly
+	RotateMonthly
+	RotateQuarterly
+	RotateYearly
```

Comment:
These const should be in the `core/secret` package, both the `domain/secret` package and the policy insertion in the DDL should import the const value from `core/secret`.

Reply:
These are strictly db artefacts, not core artefacts. They represent ids from the lookup table. Such db concerns must not leak out into core. This package is use transform the core values to the db lookup ids.

Reply:
Then how can we guarantee these two sets of const always have the correct map <->.

Reply:
See `TestRotatePolicyDBValues`
The DB is the source of truth now.